Fix editor track selection for updating tags

geichelberger opened the pull request:
Set a media package element reference on the edited track to identify it later so that the tags can be updated based on the source track.

The check if `editedTrack.getReference()` is null should not be necessary as every track should have a reference, or is that assumption wrong?

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

geichelberger committed:
Fix track selection for updating tags

Set a media package element reference on the edited track to identify it
later so that the tags can be updated based on the source track.

Arnei commented:
In this case you yourself are guaranteeing that each track has a reference by setting it yourself, right? As far as I can tell, in general there is no guarantee that an MPE has a reference, so checking for `null` is probably still advisable?

geichelberger committed:
Add warning if reference is null

If it is impossible to set the reference of the edited track, we will
log a warning that restoring the tags was skipped.
